harness/determined/common/util.py

```diff
@@ -95,6 +96,16 @@ def wrapped(*arg: Any, **kwarg: Any) -> Any:
     return wrapped
 
 
+@contextlib.contextmanager
+def preserve_random_state_context_manager(fn: Callable) -> Iterator[Callable]:
```

Comment:
Ah, given how you are using it, I'd actually skip adding this `preserve_random_state_context_manager` and calling the original decorator directly.

Like this, I mean:

```python
util.preserve_random_state(self._sync_impl)(path_info_list)
```

If you want to keep the contextmanager, have it take no arguments:

```python
with util.preserve_random_state_context_manager():
    self._sync_impl(path_list_info)
```

But passing the function in and passing it right back out isn't that helpful.